python/graphstorm/gpartition/dist_partition_graph.py

```diff
@@ -154,6 +158,23 @@ def main():
 
         logging.info("DGL graph building took %f sec", part_end - time.time())
 
+    if args.use_graphbolt:
+        dgl_version = importlib.metadata.version('dgl')
+        if version.parse(dgl_version) >= version.parse("2.0.0"):
+            # TODO: Implement distributed conversion using
+            # dgl.distributed.partition.gb_convert_single_dgl_partition()
+            logging.info("Converting partitions to GraphBolt format")
+            dgl.distributed.dgl_partition_to_graphbolt(
```

Comment:
So, do you assume the master node is going to convert partitioned graphs into graphbolt format?
Can you add a comment here that the current code assume there is a shared filesystem that the master node can access the partition result?

Reply:
Another question, what will happen if dist_partition_graph is running on SageMaker?

Reply:
Sure, I'll add a comment here to make that clear.

Generally speaking dist_partition_graph always assumes a shared FS, the way we use it in SM is by "emulating" a shared FS by pulling data locally and relying on consistent paths and file layout so that local files accesses behave like shared ones.

We haven't added SageMaker support for this PR, for SageMaker support it will be easier to use the distributed implementation, where each worker will work on its local partition. That will come in an upcoming PR.